client/src/components/PriceRangeSlider.jsx

```diff
@@ -75,19 +79,19 @@ const PriceRangeSlider = ({
         if (parseFloat(e.target.value) < minValue) {
           e.target.value = parseFloat(minValue) + 50;
           setMaxValue(e.target.value);
-          getProducts(0, subCategories, minValue, e.target.value);
+          onChange(0, subCategories, minValue, e.target.value);
         }
       }, 1000);
       debouncedMaxValue();
 
       if (parseFloat(e.target.value) > minValue) {
-        getProducts(0, subCategories, minValue, e.target.value);
+        onChange(0, subCategories, minValue, e.target.value);
```

Comment:
PriceRangeSlider should not be aware of subCategories and pageNumber. We can avoid sending subCategories to this component as prop. Instead this can be onChange(minValue, e.target.value);
And when calling this component from parent we can have:
```
<PriceRangeSlider 
  onChange={(min, max) => getProducts(0, subCategories, min, max)}
/>
```